### DISCUSSION TITLE
Sending emails with custom "From" field/domain that is not set in postal.

### DISCUSSION THREAD

**Discussion by rPawel** [postalserver/postal#1729] [Help with using Postal]
Say I have a website: `mysite.net` that I have set it up in postal and it is working great.

A client of my website who owns a website.com sets a CNAME:
`happy.website.com` that points at `happy-clients.mysite.net.`

The owner of `website.com` has added postal's IP address to its SPF record and can add DKIM record.

I want to be able to send emails on the client's behalf. I want to set `From: info@website.com` but postal stops me from doing so claiming that it does not recognise this domain. 
I can't add `website.com` or `happy.website.com` to postal as it is one of many sites that integrate with our site and such list changes.

Is there a solution workaround for this?

**Comment by miroslavskela** [comment]
You have to enable send as any on your mail server settings on postal ui, and after that you if you send using smtp you have to add sender header with mail that has domain you registered for mail server, and in from you can add custom mail address. If you send using api, than in body you can send sender property.
So if you registered postal.example.com on mail server, in sender header or property you send myemail@postal.example.com, and from mail could be custom.

**Comment by rPawel** [comment]
Thanks, @miroslavskela for the hint. I followed the instructions from https://github.com/postalserver/postal/issues/277

**Comment by rPawel** [comment]
On the side note, what is the situation with DKIM in this case? Since I haven't added the domain I don't have per domain DKIM. Is there a global one I could use?

**Comment by rPawel** [comment]
Acutaually I have praised your answer too quickly.

I have enabled the option but the emails are still failing with error.
The websites's exim that is calling postal is still being rejected whenever I try setting `From` header:
```SMTP error from remote mail server after end of data: 530 From/Sender name is not valid```
Does postal need a restart?
I can put Sender header but I need to set From.

Sample email script:
```
<?php
$to      = 'me@mysite.net';
$subject = 'Subject';
$message = 'hello';
$headers = 'Sender: info@website.com' . "\r\n" .
    'Reply-To: info@website.com' . "\r\n" .
    'From: info@website.com' . "\r\n" .
    'X-Mailer: PHP/' . phpversion();

mail($to, $subject, $message, $headers, "-f bounce@mysite.net");
```

**Comment by rPawel** [comment]
Ah the answer is:

```
<?php
$to      = 'me@mysite.net';
$subject = 'Subject';
$message = 'hello';
$headers = 'Sender: me@mysite.net' . "\r\n" .   //  <----- this needs to be my site
    'Reply-To: info@website.com' . "\r\n" .
    'From: info@website.com' . "\r\n" .
    'X-Mailer: PHP/' . phpversion();

mail($to, $subject, $message, $headers, "-f bounce@mysite.net");
```

**Comment by miroslavskela** [comment]
Yes, exactly. :)

**Comment by rPawel** [comment]
Actually this solution is still not complete as outlook marks messages with "bounce@mysite.net on behalf of info@website.com"

**Reply by miroslavskela** [reply]
Really not sure if you can do something with that. From my expirience, it's not even good to send mails on that way because it can easily be marked as spam. But let's wait for someone from postal support, they will tell you all details you need.

**Reply by rPawel** [reply]
Since client's of my website set the SPF record and are or could do DKIM I think the limitation is not in the MTA. In the end, the domain is advertising the postal server as a valid mail server. I have it set up in Exim and it is working just fine.

It seems that postal could benefit from having some white-label (bulk domain) mode, which would replicate Exim's behaviour.

Perhaps it could just take X-From and use it as From and have one default DKIM key.

**Reply by willpower232** [reply]
The only proper way to remove the "on behalf of" is to use Postal without the sender header and send email directly as the official domain. This is down to the receiving software exposing the extra information and nothing you can address otherwise IIRC.

**Reply by rPawel** [reply]
Thanks, Will, you are right. This however can't be done in my case because I'd have to configure all the domains in postal. So it is a chicken and egg situation.
